test/quoteSpec.js

```diff
@@ -0,0 +1,247 @@
+const chaiAsPromised = require('chai-as-promised')
+chai.use(chaiAsPromised)
+
+describe('ILQP', function () {
+  beforeEach(function () {
+    this.plugin = new MockPlugin()
+    // quote response
+    this.id = '85e04e5c-2357-4033-ac5b-251ce97faf33'
+    this.response = {
+      data: {
+        id: this.id,
+        method: 'quote_response',
+        data: {
+          source_amount: '1',
+          destination_amount: '1',
+          source_connector_account: 'test.example.connie',
+          source_expiry_duration: '0'
+        }
+      }
+    }
+  })
+
+  describe('quote', function () {
+    beforeEach(function () {
+      this.params = {
+        sourceAddress: 'test.example.alice',
+        destinationAddress: 'test.local.bob',
+        sourceAmount: '1',
+        sourceExpiryDuration: '0',
+        destinationExpiryDuration: '0',
+        connectors: [ 'test.example.connie' ]
+      }
+      this.result = {
+        sourceAmount: '1',
+        destinationAmount: '1',
+        connectorAccount: 'test.example.connie',
+        sourceExpiryDuration: '0'
+      }
+
+      this.plugin.sendMessage = (msg) => {
+        this.response.data.id = msg.data.id
+        this.plugin.emit('incoming_message', this.response)
+        return Promise.resolve(null)
+      }
+    })
+
+    it('should return a quote from the connector', function * () {
+      this.result.expiresAt = moment().format()
+      const response = yield ILQP.quote(this.plugin, this.params)
+      assert.deepEqual(
+        response,
+        this.result)
+    })
+
+    it('should default to getInfo\'s connectors', function * () {
+      // remove manually provided connectors
+      delete this.params.connectors
+
+      this.result.expiresAt = moment().format()
+      const response = yield ILQP.quote(this.plugin, this.params)
+      assert.deepEqual(
+        response,
+        this.result)
+    })
+
+    it('should reject if getInfo returns no connectors', function * () {
+      delete this.params.connectors
+      this.plugin.getInfo = () => ({ prefix: 'test.example.' })
+
+      yield expect(ILQP.quote(this.plugin, this.params))
+        .to.be.rejectedWith(/no connectors specified/)    
+    })
+
+    it('should quote by destination amount', function * () {
+      this.params.destinationAmount = this.params.sourceAmount
+      delete this.params.sourceAmount
+
+      this.result.expiresAt = moment().format()
+      const response = yield ILQP.quote(this.plugin, this.params)
+      assert.deepEqual(
+        response,
+        this.result)
+    })
+
+    it('should return a local quote if destination is local', function * () {
+      this.params.destinationAddress = 'test.example.bob'
+      const response = yield ILQP.quote(this.plugin, this.params)
+
+      delete this.result.connectorAccount // local quote has no connector
+      assert.deepEqual(response,
+        this.result)
+    })
+
+    it('should throw if source and dest amounts are defined', function * () {
```

Comment:
The tests should say "be rejected" instead of "throw" if that's what they do. Looking at the test output would give you a mistaken idea of what the functionality is actually expected to be (this is true of a bunch of these tests)